File: framework/languages/java/e2e/SpotService/Shared/src/main/java/systems/zlink/e2e/spotservice/shared/StateTimerHandler.java
package systems.zlink.e2e.spotservice.shared;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

import systems.zlink.framework.spots.ZLinkSpotTimerHandler;
import systems.zlink.framework.spots.ZLinkTimerTick;

public final class StateTimerHandler implements ZLinkSpotTimerHandler<UserSpot> {
    @Override
    public CompletionStage<Void> handle(
        UserSpot spot,
        ZLinkTimerTick tick) {
        spot.timerTick(tick.deliveryIndex());
        return CompletableFuture.completedFuture(null);
    }
}

File: framework/languages/java/e2e-kotlin/SpotService/Server/Play/src/main/kotlin/systems/zlink/e2e/kotlin/spotservice/play/handlers/StateTimerHandler.kt
package systems.zlink.e2e.kotlin.spotservice.play.handlers

import systems.zlink.e2e.kotlin.spotservice.Contracts
import systems.zlink.e2e.kotlin.spotservice.ScenarioState
import systems.zlink.e2e.kotlin.spotservice.play.spots.*
import systems.zlink.framework.kotlin.ZLinkSuspendingSpotTimerHandler
import systems.zlink.framework.spots.ZLinkTimerTick

class StateTimerHandler : ZLinkSuspendingSpotTimerHandler<UserSpot> {
    override suspend fun handle(spot: UserSpot, tick: ZLinkTimerTick) {
        spot.timerTick(tick.deliveryIndex())
    }
}
